### ISSUE TITLE
Question about mul(1.0-1.0/s[1]).mul(n)

### ISSUE THREAD

**Issue by albert-666** [jiecaoyu/XNOR-Net-PyTorch#41]
Hi ,when I read the last line of function 'updateBinaryGradWeight', I'm confused about 'mul(1.0-1.0/s[1]).mul(n)' . I don't understand where the coefficient come from because the progress of computing gradient has been done. It confused me. Could you tell me the reason? Thanks. 

self.target_modules[index].grad.data = m.add(m_add).mul(1.0-1.0/s[1]).mul(n)

**Comment by jiecaoyu** [comment]
Hi @albertshifengshiyu , please check [here](https://github.com/jiecaoyu/XNOR-Net-PyTorch/issues/28#issuecomment-401885963)

**Comment by albert-666** [comment]
Thank you very much! I have understand the meaning of  'mul(1.0-1.0/s[1])'. But I'm still confused about '.mul(n) is used to mitigate the effect of weight decay'.
Why is the coefficient n?  Could you explain it more detailedly? Thanks.

**Comment by jiecaoyu** [comment]
It actually does not matter what is the value. It just needs a large value, and you can replace ```n``` with some other large values like 1e9.

**Comment by albert-666** [comment]
Thank you for your patient explanation. Thanks a lot!
